File: DIYHomeAutomationAPI/DIYHomeAutomationAPI/Models/TypeDevice.cs
namespace DIYHomeAutomationAPI.Models
{
    /// <summary>
    /// This class represents a Device's Type
    /// </summary>
    public class TypeDevice
    {
        /// <summary>
        /// Device's Type Id
        /// </summary>
        public int Id { get; set; }

        /// <summary>
        /// Device's Type Name
        /// </summary>
        public string Name { get; set; } = null!;
    }
}

File: DIYHomeAutomation/app/src/main/java/com/example/diyhomeautomation/models/TypeDevice.kt
package com.example.diyhomeautomation.models

/**
 * Data class representing types of devices in DIY home automation.
 *
 * @property id The unique identifier for the device type.
 * @property name The name of the device type.
 *
 * @constructor Creates an instance of TypeDevice with the provided properties.
 */
data class TypeDevice(
    var id: Int,          // Property to store the device type ID
    var name: String      // Property to store the name of the device type
)